docker/docker_client.go

```diff
@@ -554,65 +632,88 @@ func (c *dockerClient) getBearerToken(ctx context.Context, challenge challenge,
 	return newBearerTokenFromJSONBlob(tokenBlob)
 }
 
-// detectPropertiesHelper performs the work of detectProperties which executes
-// it at most once.
-func (c *dockerClient) detectPropertiesHelper(ctx context.Context) error {
-	if c.scheme != "" {
-		return nil
+func (c *dockerClient) ping(ctx context.Context, scheme string, templateURL string, endpoint *sysregistriesv2.Endpoint, updateClient bool) error {
+	// Update the preferred endpoint to make the ping possible
+	c.preferredEndpoint = endpoint
```

Comment:
This should be documented in the struct definition of `dockerCLient`. I understand the reasoning behind doing that but it feels a bit hacky to first set it and nil it in case of an error.

Reply:
This is not part of my latest approach any more.